### PR TITLE
PR from ayush94582 forked branch

### PR DESCRIPTION
**Pull request by ayush94582**
Resolved bug #7 "Fix Previous Tip" that allows the user to navigate the tips back and forth several times as desired. 
Resolved bug #13 "No Tips in Window" by changing "tips" folder name to "Tipster"

### CODE REVIEW

File: source/Tipster.cpp [hunk 1]
```diff
@@ -387,25 +390,36 @@ Tipster::UpdateIcon(BString artwork, BString url)
 	fURL = new BString(url.String());
 }
 
-
 void
 Tipster::UpdateTip()
 {
+	//Ayush updated this method
```
Comment (humdingerb) [inline source/Tipster.cpp]: Remove comment.

File: source/Tipster.cpp [hunk 2]
```diff
@@ -430,9 +444,10 @@ Tipster::DisplayTip(BString* tip)
 void
 Tipster::DisplayPreviousTip()
 {
-	if (fPreviousTip != NULL) {
-		DisplayTip(fPreviousTip);
-
+//Ayush updated this method to incorporate multiple previous (more stable)
```
Comment (humdingerb) [inline source/Tipster.cpp]: Remove comment.

File: source/Tipster.cpp [hunk 3]
```diff
@@ -442,6 +457,8 @@ static bool
 getLocalTipsFile(entry_ref &ref, const char *language = "en")
 {
 	BStringList paths;
+     // Ayush Agarwal Bug #13
+	 //change folder name of tips to Tipster then change path below
```
Comment (humdingerb) [inline source/Tipster.cpp]: Remove comment.

File: source/Tipster.cpp [hunk 4]
```diff
@@ -3,7 +3,7 @@
  * All rights reserved. Distributed under the terms of the MIT license.
  */
 #include "Tipster.h"
-
+#include <iostream>
```
Comment (humdingerb) [inline source/Tipster.cpp:8]: Add new line above and below.

File: source/Tipster.cpp [hunk 5]
```diff
@@ -3,7 +3,7 @@
  * All rights reserved. Distributed under the terms of the MIT license.
```
Comment (humdingerb) [inline source/Tipster.cpp:4]: You may add your copyright in the same format under Vale's.

**Review by humdingerb — CHANGES_REQUESTED** [review]
Pleas also read https://www.haiku-os.org/development/coding-guidelines and correct the few whitespace issues.

File: source/Tipster.cpp [hunk 6]
```diff
@@ -1,9 +1,12 @@
 /*
  * Copyright 2015 Vale Tolpegin <valetolpegin@gmail.com>
+ * Copyright 2016 Ayush Agarwal <ayush94582@yahoo.com>
  * All rights reserved. Distributed under the terms of the MIT license.
  */
 #include "Tipster.h"
+
 #include <iostream>
```
Comment (humdingerb) [inline source/Tipster.cpp:8]: Sorry, I missed that before: <iostream> fits better farther down before <stdio.h>

File: source/Tipster.cpp [hunk 7]
```diff
@@ -52,6 +55,7 @@ Tipster::Tipster()
 	fURL = new BString("");
 	fArtworkTitle = new BString("");
 	fPreviousTip = new BString("");
+	ind=0;
```
Comment (humdingerb) [inline source/Tipster.cpp]: Coding style: add spaces before and after "=".
Rename "ind" to something more descriptive.

File: source/Tipster.cpp [hunk 8]
```diff
@@ -397,15 +402,26 @@ Tipster::UpdateTip()
 
 		fTipsList.Remove(0);
 	}
+	
+	if(ind==storepast.CountStrings()-1){
```
Comment (humdingerb) [inline source/Tipster.cpp]: Coding style: space after "if", around "==", around "-", before "{".

File: source/Tipster.cpp [hunk 9]
```diff
@@ -397,15 +402,26 @@ Tipster::UpdateTip()
...
+		DisplayTip(new BString(fTipsList.StringAt(fTipNumber)));
 
-	DisplayTip(new BString(fTipsList.StringAt(fTipNumber)));
+		fPreviousTip = new BString(fCurrentTip->String());
+		fCurrentTip = new BString(fTipsList.StringAt(fTipNumber));
 
-	fPreviousTip = new BString(fCurrentTip->String());
-	fCurrentTip = new BString(fTipsList.StringAt(fTipNumber));
-	fTipsList.Remove(fTipNumber);
+		ind=storepast.CountStrings();
```
Comment (humdingerb) [inline source/Tipster.cpp]: Coding style: space around "=".

File: source/Tipster.cpp [hunk 10]
```diff
@@ -397,15 +402,26 @@ Tipster::UpdateTip()
...
+		storepast.Add(fCurrentTip->String());
 
+		fTipsList.Remove(fTipNumber);
+	}
+
```
Comment (humdingerb) [inline source/Tipster.cpp:418]: remove blank line

File: source/Tipster.cpp [hunk 11]
```diff
@@ -397,15 +402,26 @@ Tipster::UpdateTip()
...
-	fPreviousTip = new BString(fCurrentTip->String());
-	fCurrentTip = new BString(fTipsList.StringAt(fTipNumber));
-	fTipsList.Remove(fTipNumber);
+		ind=storepast.CountStrings();
+		storepast.Add(fCurrentTip->String());
 
+		fTipsList.Remove(fTipNumber);
+	}
+	
+	else{
```
Comment (humdingerb) [inline source/Tipster.cpp]: space after "else"

File: source/Tipster.cpp [hunk 12]
```diff
@@ -442,6 +458,7 @@ static bool
 getLocalTipsFile(entry_ref &ref, const char *language = "en")
 {
 	BStringList paths;
+
```
Comment (humdingerb) [inline source/Tipster.cpp]: remove blank line

**Review by humdingerb — CHANGES_REQUESTED** [review]
re-read the style guide...

**Comment by scottmc** [comment]
Any updates on this?

**Review by scottmc — DISMISSED** [review]

**Comment by scottmc** [comment]
any idea on how to proceed of this?

**Comment by pulkomandy** [comment]
The conflicts should be solved and I still see some style problems (wrong indentation style in at least one place, I didn't do a very careful review). These should be addressed, preferably by the original submitter, but if he disappeared and isn't interested, we will have to do it ourselves...

**Comment by AkshayAgarwal007** [comment]
Apart from the conflicts these are some issues that are there:
1. the list 'storepast' would store the entire history of tips that are displayed till a certain point. I think having the history longer than the total no. of tips that are there is of no good. (given tips are displayed randomly but all of them are displayed)
2. Everytime 'fTipsList' becomes empty (after all the tips are displayed) loading of the tips again from the file will take place. 
3. Moving to next and previous tip by incrementing/decrementing tipindex hasn't been implemented properly.

**Comment by pulkomandy** [comment]
There is now another pull request (#17) that solves this in, I think, a much cleaner way.

**Comment by scottmc** [comment]
Closing this one then...

**Pull request closed without merging**